service/nodeconfig/v1/resource/node/update.go

```diff
@@ -6,15 +6,15 @@ import (
 	"github.com/giantswarm/operatorkit/framework"
 )
 
+// ApplyUpdateChange is a noop. Draining nodes does not require more complex
+// CRUD management of resources. See ApplyCreateChange for the business logic
+// implementation.
 func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
 	return nil
 }
 
 func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
 	patch := framework.NewPatch()
+	patch.SetCreateChange(true) // hack to execute ApplyCreateChange
```

Comment:
I don't consider it a hack. It's designed to make that possible.

Reply:
Fair. I just wanted to have a comment to explain what is happening and why. While we are here I want to share another thought about this. Our framework and resource interface is designed for more complex CRUD kind of implementations. The node-operator is rather simple and only cares about one event and then it has to implement a series of actions. There is no real current and desired state and pressing the node-operator's functionality into the resource interface does not make that much sense. See for instance all the "noop" comments. I think it would be cool to have another interface for such simple purposes. One option would be to have a wrapper for the typical resource that only provides a "callback" for a create event (or another). We could sort this in other ways as well but then we would probably have to use a different framework implementation as well. Uncertain. WDYT?

Reply:
Yes, I was thinking about it for a long time. I was thinking about an interface like:

```
type Resource interface {
	EnsureCreated(obj interface{}) error
	EndureDeleted(obj interface{}) error
}
```

This is the same interface as CRDClient has.

Then wie can add some kind of intermediate struct like CRUDResource which can translate the complex interface to the simple one.

Reply:
Something like this would be cool. This interface would just be too complex already for the node-operator since it only cares about the create event. There are probably lots of different incarnations of such interfaces depending on the use case the operator cares about.

Reply:
We are not far enough with the operatorkit changes. I go ahead and get the refactoring done later. I just want to make progress here now.